src/screens/PrivateChat/ContactsPicker/index.js

```diff
@@ -0,0 +1,253 @@
+
+const DONE_BUTTON = 'DONE_BUTTON';
+const DISABLED_RIGHT_BUTTON = {
+  title: 'Done',
+  id: DONE_BUTTON,
+  buttonColor: Colors.navigationButtonColor,
+  disabled: true,
+};
+const ADD_CONTACT_ERROR = 'Error adding contact, please try again later.';
+
+type Props = {
+  /**
+   * @desc React Native Navigation navigator object.
+   */
+  navigator: Navigator,
+  /**
+   * @desc List of all contacts.
+   */
+  contacts: Array<Contact>,
+  /**
+   * @desc List of initially selected contacts.
+   */
+  initialSelectedContacts: Array<Contact>,
+  /**
+   * @desc Function to add a new contact
+   * @param {string} identityKey Identity key of user.
+   * @param {function} callback Callback
+   */
+  addContact: (identityKey: string, callback: (error: Error | null) => void) => void,
+  /**
+   * @desc Callback function to call when submitting contact selection.
+   */
+  onContactsSelected: (selectedContacts: Array<Contact>) => void,
+};
+
+type State = {
+  /**
+   * @desc The identityKey of the added contact.
+   */
+  addedContactIdentityKey: string,
+  /**
+   * @desc Flag whether loading is in progress.
+   */
+  loading: boolean,
+  /**
+   * @desc Error message for adding contact failure.
+   */
+  addContactError: string,
+  /**
+   * @desc Flag whether done button is disabled.
+   */
+  doneBtnDisabled: boolean,
+};
+
+class ContactsPickerScreen extends NavigatorComponent<Props, State> {
+  static navigatorButtons = {
+    leftButtons: [],
+    rightButtons: [DISABLED_RIGHT_BUTTON],
+  };
+
+  static defaultProps = {
+    initialSelectedContacts: [],
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      addedContactIdentityKey: '',
+      loading: false,
+      addContactError: '',
+      doneBtnDisabled: true,
+    };
+  }
+
+  selectize: Selectize;
+
+  onNavBarButtonPress(id) {
+    if (id === DONE_BUTTON) {
+      const selectedContacts = this.selectize.getSelectedItems().result;
+      this.props.onContactsSelected(selectedContacts);
+    }
+  }
+
+  componentDidMount() {
+    if (this.props.initialSelectedContacts.length > 0) {
+      this.enableDoneButton();
+
+      this.props.initialSelectedContacts.forEach((contact) => {
+        this.selectize._selectItem(contact.profile.identityKey);
+      });
+    }
+  }
+
+  componentDidUpdate() {
+    const addedContact = _.find(
+      this.props.contacts,
+      contact => contact.profile.identityKey === this.state.addedContactIdentityKey,
+    );
+
+    if (addedContact) {
+      this.selectize._selectItem(addedContact.profile.identityKey);
+      this.setState({ addedContactIdentityKey: '' });
+      this.enableDoneButton();
+    }
+  }
+
+  enableDoneButton = () => {
+    if (this.state.doneBtnDisabled) {
+      this.props.navigator.setButtons({
+        rightButtons: [{
+          ...DISABLED_RIGHT_BUTTON,
+          disabled: false,
+        }],
+      });
+      this.setState({ doneBtnDisabled: false });
```

Comment:
Don't you want to make the state driving it? Currently, the flow is:
 - someone calls function `enable`/`disable`
 - button changed
 - state updated.

More 'React' way would be to implement it like:
 - someone calls setState({ doneButtonDisabled: true })
 - on componentDidUpdate if there is a difference between old and new `doneButtonDisabled ` flag, change the state of navigation buttons.